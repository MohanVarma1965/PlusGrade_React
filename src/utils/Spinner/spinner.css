@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.loader {
  border: 8px solid #f3f3f3;
  border-top: 8px solid #3498db;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  animation: spin 2s linear infinite;
  z-index: 1001;
}

/* Styles to center the spinner and add overlay */
.center-spinner {
  position: fixed; /* Fixed position */
  top: 0;
  left: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.5); /* Black with 50% opacity */
  z-index: 1000; /* Ensure overlay is above the page content but below the loader */
}
