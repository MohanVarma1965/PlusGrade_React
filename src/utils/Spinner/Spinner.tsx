import React from "react";
import "./spinner.css";

const Spinner: React.FC = () => {
  return (
    <div
      className="center-spinner"
      data-testid="spinner-element"
      role="progressbar"
      aria-busy="true"
      aria-label="Loading..."
    >
      <div className="loader"></div>
    </div>
  );
};

export default Spinner;
