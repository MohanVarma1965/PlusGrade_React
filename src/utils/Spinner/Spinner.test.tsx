import { render } from "@testing-library/react";
import Spinner from "./Spinner";

// Testing the Spinner component
describe("Case 1. Spinner Component", () => {
  it("1.1 Renders without crashing", () => {
    render(<Spinner />);
  });

  it("1.2 Renders the spinner element", () => {
    const { getByTestId } = render(<Spinner />);
    const spinnerElement = getByTestId("spinner-element");
    expect(spinnerElement).toBeInTheDocument();
  });

  it("1.3 Renders the loader inside the spinner", () => {
    const { container } = render(<Spinner />);
    const loaderElement = container.querySelector(".loader");
    expect(loaderElement).toBeTruthy();
  });

  it("1.4 Has accessibility label", () => {
    const { getByLabelText } = render(<Spinner />);
    expect(getByLabelText("Loading...")).toBeInTheDocument();
  });

  it("1.5 matches snapshot", () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
