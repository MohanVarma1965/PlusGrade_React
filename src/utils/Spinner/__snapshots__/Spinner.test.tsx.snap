// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1. Spinner Component 1.5 matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="Loading..."
    class="center-spinner"
    data-testid="spinner-element"
    role="progressbar"
  >
    <div
      class="loader"
    />
  </div>
</DocumentFragment>
`;
